### PR TITLE
Tagging update

### PR DESCRIPTION
**Pull request by jtcul**
*Description of changes:*
Migrate to the new "tagging" attribute. Original PR: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-kms/pull/49. I have addressed my own comments and raised a new PR.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sruthisrip — APPROVED** [review]

File: common/src/main/java/software/amazon/kms/common/TagHelper.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+     * generateTagsForCreate
+     *
+     * Generate tags to put into resource creation request.
+     */
+    public final Map<String, String>  generateTagsForCreate(final ResourceHandlerRequest<M> handlerRequest) {
+        final Map<String, String> tagMap = new HashMap<>();
+
+        // merge system tags with desired resource tags if your service supports CloudFormation system tags
+        // KMS does not support system tags currently.
+        // tagMap.putAll(handlerRequest.getSystemTags());
```
Comment (nigamaviral) [inline common/src/main/java/software/amazon/kms/common/TagHelper.java:63]: can we remove this for now and add it later when we actually support it ?

File: common/src/main/java/software/amazon/kms/common/TagHelper.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        if (handlerRequest.getDesiredResourceTags() != null) {
+            tagMap.putAll(handlerRequest.getDesiredResourceTags());
+        }
+        // TODO: get tags from resource model based on your tag property name.
+        // TODO: tagMap.putAll(convertToMap(resourceModel.getTags()));
+        // getDesiredResourceTags() gets both resource and stack level tags. getTags() gets resource level tags only.
```
Comment (nigamaviral) [inline common/src/main/java/software/amazon/kms/common/TagHelper.java:69]: why is this in commented state ?

File: common/src/main/java/software/amazon/kms/common/TagHelper.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+     * we will get previous attached user defined tags from both handlerRequest.getPreviousResourceTags (stack tags)
+     * and handlerRequest.getPreviousResourceState (resource tags).
+     */
+    public Map<String, String> getPreviouslyAttachedTags(final ResourceHandlerRequest<M> handlerRequest) {
+        // get previous stack level tags from handlerRequest
+        final Map<String, String> previousTags = handlerRequest.getPreviousResourceTags() != null ?
+                handlerRequest.getPreviousResourceTags() : Collections.emptyMap();
+        // TODO: get resource level tags from previous resource state based on your tag property name
+        // TODO: previousTags.putAll(handlerRequest.getPreviousResourceState().getTags());
+        // getDesiredResourceTags() gets both resource and stack level tags. getTags() gets resource level tags only.
```
Comment (nigamaviral) [inline common/src/main/java/software/amazon/kms/common/TagHelper.java:98]: ditto

File: common/src/main/java/software/amazon/kms/common/TagHelper.java [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+     * we will get new user defined tags from both resource model and previous stack tags.
+     */
+    public Map<String, String> getNewDesiredTags(final ResourceHandlerRequest<M> handlerRequest) {
+        // get new stack level tags from handlerRequest
+        final Map<String, String> desiredTags = handlerRequest.getDesiredResourceTags() != null ?
+                handlerRequest.getDesiredResourceTags() : Collections.emptyMap();
+
+        // TODO: get resource level tags from resource model based on your tag property name
+        // TODO: desiredTags.putAll(convertToMap(resourceModel.getTags()));
+        // getDesiredResourceTags() gets both resource and stack level tags. getTags() gets resource level tags only.
```
Comment (nigamaviral) [inline common/src/main/java/software/amazon/kms/common/TagHelper.java:115]: ditto

File: common/src/main/java/software/amazon/kms/common/TagHelper.java (hunk 1 above)
Comment (jtcul) [reply common/src/main/java/software/amazon/kms/common/TagHelper.java:63]: Yes, will remove it. These comments got copy-pasted from a resource agnostic sample.

File: common/src/main/java/software/amazon/kms/common/TagHelper.java (hunk 2 above)
Comment (jtcul) [reply common/src/main/java/software/amazon/kms/common/TagHelper.java:69]: Yes, see above comment, will remove

**Review by nigamaviral — APPROVED** [review]

**Pull request merged**